From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: David Llewellyn-Jones <david@flypig.co.uk>
Date: Sun, 26 Nov 2023 12:49:29 +0000
Subject: [PATCH 34/67] Supress URLQueryStrippingListService.jsm error

The URLQueryStrippingListService service is initialised by the
"profile-after-change" event. The event also goes through to the
observer, which is coded to generate an error when the event is received
("unexpected event profile-after-change"). This change supressed the
error as long as the event is received before the service has been
initialised (i.e. it's the event that's caused the service to be
initialised).

It's not clear why this isn't an issue upstream.

See the following for context:

https://bugzilla.mozilla.org/show_bug.cgi?id=1706608

https://phabricator.services.mozilla.com/D117376

https://bugzilla.mozilla.org/show_bug.cgi?id=1723981

Co-authored-by: Raine Makelainen <raine.makelainen@jolla.com>
---
 .../antitracking/URLQueryStrippingListService.sys.mjs        | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/toolkit/components/antitracking/URLQueryStrippingListService.sys.mjs b/toolkit/components/antitracking/URLQueryStrippingListService.sys.mjs
index e9be0a3ac405..4a2f6447a342 100644
--- a/toolkit/components/antitracking/URLQueryStrippingListService.sys.mjs
+++ b/toolkit/components/antitracking/URLQueryStrippingListService.sys.mjs
@@ -278,6 +278,11 @@ export class URLQueryStrippingListService {
         let prefValue = Services.prefs.getStringPref(data, "");
         this._onPrefUpdate(data, prefValue);
         break;
+      case "profile-after-change":
+        if (this.#isInitialized) {
+          console.error(`Unexpected post-init event ${topic}`);
+        }
+        break;
       default:
         console.error(`Unexpected event ${topic}`);
     }
